src/reanimated2/layoutReanimation/web/animationsManager.ts

```diff
@@ -141,17 +122,46 @@ export function startWebLayoutAnimation<
     initialAnimationName as AnimationNames
   );
 
-  if (checkReduceMotionFail(animationConfig, hasEnteringAnimation, element)) {
-    return;
+  if (checkReduceMotionFail(animationConfig)) {
+    return null;
   }
 
-  chooseAction(
-    animationType,
-    animationConfig,
-    element,
-    transitionData as TransitionData,
-    transform
+  return { animationConfig, transform };
+}
+
+export function startWebLayoutAnimation<
+  ComponentProps extends Record<string, unknown>
+>(
+  props: Readonly<AnimatedComponentProps<ComponentProps>>,
+  element: HTMLElement,
+  animationType: LayoutAnimationType,
+  shouldMakeVisible = false,
+  transitionData?: TransitionData
+) {
+  let visibilityDelay = 0;
+
+  const maybeAnimationConfigWithTransform = tryGetAnimationConfigWithTransform(
+    props,
+    animationType
   );
+
+  if (maybeAnimationConfigWithTransform) {
```

Comment:
It seems that `tryGetAnimationConfigWithTransform` should return `null` when it is not the first render and `animationType` is `entering`, in order to avoid unnecessary code execution.